### PR TITLE
Add Domains API endpoints to openapi spec. OKTA-387064

### PR DESCRIPTION
**Pull request by serhiibuniak-okta**

### CODE REVIEW

File: dist/spec.json [hunk 1]
```diff
@@ -13555,6 +13793,159 @@
...
+      },
+      "x-okta-tags": [
+        "Domain"
+      ]
+    },
+    "DomainListResponse": {
+      "properties": {
+        "domains": {
+          "items": {
+            "$ref": "#/definitions/DomainResponse"
```
Comment (anitadongare-okta) [inline dist/spec.json:13905]: for `DomainListReponse` items, I think we will need to have a separate definition for the array item. 

It is currently defined as` DomainResponse` which contains all the domain properties and is returned when we do a `Get` for single domain. 
Whereas for `Get All` we will respond with a subset of properties for each domain - https://developer.okta.com/docs/reference/api/domains/#domainlistresponse-properties

Comment (anitadongare-okta) [reply dist/spec.json:13905]: Please ignore above comment, I think it looks correct as it is aligned with the implementation. I will try to update the docs to reflect the same.

**Review by anitadongare-okta — APPROVED** [review]
LGTM 🚢 thanks @serhiibuniak-okta!

**Review by bryanapellanes-okta — APPROVED** [review]
I don't see any issues.

**Pull request merged**